core/vm/instructions.go

```diff
@@ -259,7 +260,11 @@ func opAddress(pc *uint64, interpreter *EVMInterpreter, scope *ScopeContext) ([]
 func opBalance(pc *uint64, interpreter *EVMInterpreter, scope *ScopeContext) ([]byte, error) {
 	slot := scope.Stack.peek()
 	address := common.Address(slot.Bytes20())
-	slot.SetFromBig(interpreter.evm.StateDB.GetBalance(address))
+	balance, err := interpreter.evm.StateDB.GetBalance(address)
+	if err != nil {
+		return nil, err
```

Comment:
Digging into this, I don't think its safe to return errors from these functions. I think we need to signal to the VM that an operation failed through. Its not obvious to do here, because `opBalance` as previously infallible, but if you look at some of the fallible opcodes (e.g. `opCall`) you can see they push a return value onto the stack AND return that value without error. I think we need to do that here and everywhere anywhere else we make an opcode fallible

Reply:
I think these operations need to use the `reverts` or `returns` flag in the operation descriptor:
```
type operation struct {
	// execute is the operation function
	execute     executionFunc
	constantGas uint64
	dynamicGas  gasFunc
	// minStack tells how many stack items are required
	minStack int
	// maxStack specifies the max length the stack can have for this operation
	// to not overflow the stack.
	maxStack int

	// memorySize returns the memory size required for the operation
	memorySize memorySizeFunc

	halts   bool // indicates whether the operation should halt further execution
	jumps   bool // indicates whether the program counter should not increment
	writes  bool // determines whether this a state modifying operation
	reverts bool // determines whether the operation reverts state (implicitly halts)
	returns bool // determines whether the operations sets the return data content
}
```

This is how the VM knows to check the status of the code and return or revert execution.